include/nbl/builtin/hlsl/scan/descriptors.hlsl

```diff
@@ -17,21 +17,21 @@ namespace hlsl
 namespace scan
 {
 
-template<uint32_t scratchElementCount=SCRATCH_SZ - 1> // (REVIEW): This should be externally defined. Maybe change the scratch buffer to RWByteAddressBuffer? Annoying to manage though...
+template<uint32_t scratchElementCount=SCRATCH_SZ - 1>
 struct Scratch
 {
     uint32_t workgroupsStarted;
     uint32_t data[scratchElementCount];
 };
 
-[[vk::binding(0 ,0)]] RWStructuredBuffer<uint32_t /*Storage_t*/> scanBuffer; // (REVIEW): Make the type externalizable. Decide how (#define?)
+[[vk::binding(0 ,0)]] RWStructuredBuffer<Storage_t> scanBuffer; // (REVIEW): Make the type externalizable. Decide how (#define?)
 [[vk::binding(1 ,0)]] RWStructuredBuffer<Scratch> /*globallycoherent (seems we can't use along with VMM)*/ scanScratchBuf; // (REVIEW): Check if globallycoherent can be used with Vulkan Mem Model
```

Comment:
this one you need to use `BDA` 

globallycoherent can't be used with VMM, but DXc doesn't support/emit/upgrade to VMM

IIRC you can mark individual load/store as coherent even before VMM (just no acquire/release cause thats VMM) with spir-v intrinsics and that should be enough

Also because the scratch needs to be coherent, it only makes sense to come from a buffer, and you might as well use BDA for it

Reply:
Even without VMM you can use Release/Acquire/SeqCst

The only difference between pre and post VMM, is because Volatile used to be a Memory Operand, and in VMM its a MEmory Semantic.

https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_memory_semantics_id